### ISSUE TITLE
[suggest]  add git submodule support for folly/fizz

### ISSUE THREAD

**Issue by fawdlstty** [facebookincubator/mvfst#86]

**Comment by yangchi** [comment]
It's a good idea. Thank you! We will definitely welcome such PR if you'd like to give it a try.

(The source control and other workflow inside FB is quite different. This is the sort of pain the OSS users feel but we don't. So this likely will be very low pri for us to work on.)

**Comment by fawdlstty** [comment]
I don't know cmake. This is very important for submodules

**Comment by odelalleau** [comment]
I also just ran into compilation issues related to the lack of support for submodules (or any other way to ensure the folly & fizz dependencies are synched to their expected version).

I just realized though that these versions seem to be defined within build/deps/github_hashes/ so it shouldn't be too hard to parse these files in build_helper.sh so as to checkout the correct commit for each dependency. This seems to me like a less intrusive change (vs. moving to submodules), so I'll go ahead with it and submit a PR once I have something that works.

**Comment by lnicco** [comment]
yeah the proxygen build does that. I'll sync them so that mvfst does that too.

**Comment by odelalleau** [comment]
Looks like we both did a similar thing in parallel ;) I still kept #94 up if you could please have a look.

I noticed when updating to latest version that a new dependency was add (fmt) so a similar issue still exists for that one. Unfortunately it doesn't look like there's an associated revision under build/deps for that one... is it coming or is there a need for another synch logic?
